extensions/amp-geo/0.1/amp-geo.js

```diff
@@ -207,6 +215,115 @@ export class AmpGeo extends AMP.BaseElement {
         'GEONOTPATCHED: amp-geo served unpatched, ISO country not set'
       );
     }
+
+    return this.mode_ !== mode.GEO_API
+      ? Promise.resolve()
+      : this.fetchCountry_().then(country => {
+          if (country) {
+            this.country_ = country;
+          } else {
+            // if API request fails, leave the country at the default 'unknown'
+            this.error_ = true;
+            dev().error(
+              TAG,
+              'GEONOTPATCHED: amp-geo served unpatched and API response not valid, ISO country not set'
+            );
+          }
+        });
+  }
+
+  /**
+   * Ensure API URL definition is usable and cast its type
+   * @param {*} url
+   * @return {?string}
+   */
+  validateApiUrl(url) {
+    if (typeof url !== 'string') {
+      user().error(TAG, 'geoApiUrl must be a string URL');
+      return null;
+    }
+
+    if (!isSecureUrlDeprecated(url)) {
```

Comment:
Hmmm the function name tells me that it's deprecated : )
`#isSecureUrlDeprecated()` doesn't handle the relative url case `//`, but `#assertHttpsUrl()` does.
The `elementContext` and `sourceName` makes the `#assertHttpsUrl()` looks like it's tied to AMP elements. But they are used only as error message. A better name is probably `context` instead of `elementContext`. What about
```
assertHttpsUrl(url, 'geoApiUrl', 'AMP_CONFIG urls')
```

Reply:
I'm not convinced `assertHttpsUrl()` is the right choice. It's a wrapper for `isSecureUrlDeprecated(url) || /^(\/\/)/.test(url)`. Given the privacy implications of user location, do you really want to allow relative protocol? It seems more reasonable to directly make use of `isSecureUrlDeprecated` which requires HTTPS even if an AMP page is served from HTTP.

Reply:
The reason it's deprecated is because it doesn't work properly in the different ampdoc modes. Use `Services.urlFor`'s `isSecure()` instead.